src/core/qgsvectorlayer.h

```diff
@@ -921,6 +921,17 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     long featureCount( const QString &legendKey ) const;
 
+    /**
+     * Determines if this vector layer is empty.
+     * A layer is considered empty if either the data provider or
+     * the edit buffer have features in them.
+     * This means, in case the data provider contains features which have all
```

Comment:
This is a bit unclear- it took me a couple of reads to understand. Maybe this is better:

```
\warning A layer is considered empty if neither the data provider or the edit buffer have features in them. This means the returned value may be inaccurate for layers with unsaved changes, e.g. in the case of all features from the data provider having been deleted within the edit buffer"

Reply:
`s/e.g./i.e./` and good to go